addons/Dexie.Observable/src/Dexie.Observable.js

```diff
@@ -242,7 +242,7 @@ function Observable(db) {
                 });
             }
             // Add new sync node or if this is a reopening of the database after a close() call, update it.
-            return db.transaction('rw', '_syncNodes', () => {
+            return Dexie.ignoreTransaction(() => {
```

Comment:
Could we keep the transaction here to make the addition of syncNode atomic but just add Dexie.ignoreTransaction() around it?

```js
return Dexie.ignoreTransaction(() => db.transaction('rw', '_syncNodes', () => {
    ...
}));

Reply:
Alas not - it seems like some of the logic within the existing `transaction` block still causes the transaction to become inactive, even when it's wrapped (rather than replaced) by `ignoreTransaction`.

Is it possible that the [`db._syncNodes.where` query](https://github.com/dfahlander/Dexie.js/tree/77f0b08c58784bfaaf9e2f5e26dd4e1d7b7d3094/addons/Dexie.Observable/src/Dexie.Observable.js#L240-L242) closes the transaction early, for some reason?

From debugging a bit more, it seems that any operations on `db._syncTables` (either read or write) within the `rw` transaction and after the `where` query triggers the `TransactionInactiveError`.

Reply:
NB: the requirement to keep the atomic behaviour makes sense; I'm trying a few other approaches at the moment to see whether it's possible to retain the transaction.